ci/jenkins/jobs/macros.yaml

```diff
@@ -78,14 +78,35 @@
     name: builder-workload-cluster-cleanup
     builders:
       - shell: |-
+          #!/bin/bash
           cluster="${CLUSTERNAME}"
 
           echo '=== Clean up cluster ==='
           export KUBECONFIG="/var/lib/jenkins/kubeconfig-mgmt"
-          kubectl delete Machine ${cluster}-controlplane-0 -n ${cluster} || true
-          kubectl delete MachineDeployment ${cluster}-md-0 -n ${cluster} || true
-          kubectl delete cluster ${cluster} -n ${cluster} || true
-          kubectl delete ns ${cluster} || true
+
+          kubectl delete --ignore-not-found=true Machine ${cluster}-controlplane-0 -n ${cluster}
+          kubectl delete --ignore-not-found=true MachineDeployment ${cluster}-md-0 -n ${cluster}
+
+          retry=12
+          while [ "${retry}" -gt 0 ]; do
+            echo '=== Waiting for Machines to be deleted ==='
+            machines="$(kubectl get machine -n ${cluster} --no-headers=true 2>/dev/null | wc -l)"
```

Comment:
Out of curiosity, does not `kubectl delete Machine` wait for the object to be gone before returning? I think the default behavior is waiting without timeout.